### PR TITLE
Implementing .size and groupby size method (#1627)

### PR DESCRIPTION
**Pull request by JamesJeffryes**
Note here that I had to amend _apply_chunk to accept columns=None to match the pandas result.

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -253,6 +253,10 @@ def npartitions(self):
         return len(self.divisions) - 1
 
     @property
+    def size(self):
+        return self.reduction(lambda x: x.size, np.sum, token='size', meta=int)
```
Comment (jcrist) [inline dask/dataframe/core.py]: In an effort to keep lambda's out of non-user defined graphs, can you move this functionality to the `methods.py` file, as done here: https://github.com/dask/dask/blob/master/dask/dataframe/methods.py#L73-L74?

File: dask/dataframe/tests/test_dataframe.py [hunk 2]
```diff
@@ -662,6 +662,11 @@ def test_len():
     assert len(d.a) == len(full.a)
 
 
+def test_size():
+    assert eq(d.size, full.size)
+    assert eq(d.a.size, full.a.size)
```
Comment (jcrist) [inline dask/dataframe/tests/test_dataframe.py]: Can you add a test for `d.index.size` as well?

**Comment by jcrist** [comment]
Thanks for the fix! Overall this looks good to me. I'm not sure if the additional check in `_apply_chunk` is the best way to deal with this, but it's simple and all tests pass. I left a few nitpicky comments, happy to merge once those are addressed. You can ignore the appveyor failures for now, the issue there is unrelated.

**Comment by mrocklin** [comment]
Thanks @JamesJeffryes !

**Pull request merged**